File: medium/0080_remove_duplicates_from_sorted_array_ii/solution.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if len(nums) <= 2:
            return len(nums)

        insert = 2

        for read in range(2, len(nums)):
            if (
                nums[read] != nums[insert - 1]
                or nums[read] != nums[insert - 2]
            ):
                nums[insert] = nums[read]
                insert += 1

        return insert

File: medium/0080_remove_duplicates_from_sorted_array_ii/solution.cpp
#include <vector>

class Solution {
public:
  int removeDuplicates(std::vector<int> &nums) {
    const size_t n = nums.size();

    size_t insert{};
    for (size_t read{}; read < n; ++insert) {
      nums[insert] = nums[read++];

      while (read + 1 < n && nums[read + 1] == nums[insert])
        ++read;
    }

    return static_cast<int>(insert);
  }
};
